js/layout-javascript/agenda-code.js

```diff
@@ -1247,10 +1246,10 @@ DynamicList.prototype.prepareToRenderLoop = function(rows) {
     return row[dateField];
   }));
 
-  _this.modifiedRecords = newRecords;
+  _this.agendasByDay = newRecords;
 }
 
-DynamicList.prototype.renderLoopHTML = function(records) {
+DynamicList.prototype.renderLoopHTML = function(iterateeCb, finishCb) {
```

Comment:
@angelnikolov Why not use a Promise? We haven't been using callbacks for a while. Most of our code has moved on to use Promises.

Reply:
I didn't want to mix up between iterative callback and a promise so the API is more obvious. I can easily change that to a Promise if everyone agrees but I think it's cleaner this way.

Reply:
I see.

I think it's best to use a Promise instead of the `finishCb`. The `iteratee` can be kept as a parameter. This way when `.renderLoopHTML()` can use `.then()` and `.catch()` for errors and so on. We've been relying on that for many things.

It'll be odder to change to a `finishCb` parameter half way through a bunch of `.then()` chains.

Reply:
Okay, will change those to promises.

Reply:
Done, @tonytlwu would you mind checking it again?:) Thanks!